src/main/java/org/openrewrite/java/spring/boot2/IntegrationSchedulerPoolRecipe.java

```diff
@@ -0,0 +1,135 @@
+//                    }
+//                    if (boot25 && si) {
+//                        break;
+//                    }
+//                }
+//                if (boot25 && si) {
+//                    return maven.withMarkers(maven.getMarkers().searchResult());
+//                }
+//                return maven;
+//            }
+//        };
+//    }
+
+    @Override
+    protected List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx) {
+        Set<Path> paths = new HashSet<>();
+        MavenVisitor visitor = new MavenVisitor() {
+            @Override
+            public Maven visitMaven(Maven maven, ExecutionContext ctx) {
+                Collection<Pom.Dependency> deps = maven.getModel().getDependencies(Scope.Compile);
+                boolean boot25 = false;
+                boolean si = false;
+                for (Pom.Dependency d : deps) {
+                    if (!boot25) {
+                        boot25 = d.getArtifactId().equals("spring-boot")
+                                && d.getVersion().startsWith("2.5.")
+                                && d.getGroupId().equals("org.springframework.boot");
+                    }
+                    if (!si) {
+                        si = d.getArtifactId().equals("spring-integration-core")
+                                && d.getGroupId().equals("org.springframework.integration");
+                    }
+                    if (boot25 && si) {
+                        Path projectDir = maven.getSourcePath().getParent();
+                        paths.addAll(maven.getResources(projectDir, ctx));
+                        break;
+                    }
+                }
+                return maven;
+            }
+        };
+
+        for (SourceFile f : before) {
+            visitor.visit(f, ctx);
+        }
+
+        List<SourceFile> modified = ListUtils.map(before, source -> (SourceFile) new PropertiesVisitor<ExecutionContext>() {
+            @Override
+            public Properties visitFile(Properties.File file, ExecutionContext ctx) {
+                if ("application.properties".equals(file.getSourcePath().getFileName().toString())
+                        && paths.contains(file.getSourcePath())) {
+                    Properties.File p = (Properties.File) super.visitFile(file, ctx);
+                    Properties.Entry schedulerPoolEntry = getCursor().pollMessage(SCHEDULER_PROPERTY_ENTRY);
+                    int idx = schedulerPoolEntry != null ? p.getContent().indexOf(schedulerPoolEntry) : -1;
+                    List<Properties.Content> contents = new ArrayList<>(p.getContent());
+                    Properties.Comment comment = new Properties.Comment(Tree.randomId(), "\n", Markers.EMPTY, MIGRATION_MESSAGE);
+                    if (idx >= 0) {
+                        contents.add(idx, comment);
+                    } else {
+                        contents.add(comment);
+                    }
+                    p = p.withContent(contents);
```

Comment:
consider using `ListUtils#insertAll` and `ListUtils#concat` for these list modifications

Reply:
Aha, thanks, will consider this of course!